src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -1179,7 +1179,7 @@ private void getMethodSig(CORINFO_METHOD_STRUCT_* ftn, CORINFO_SIG_INFO* sig, CO
             Get_CORINFO_SIG_INFO(method, sig: sig, scope: null);
         }
 
-        private bool getMethodInfo(CORINFO_METHOD_STRUCT_* ftn, CORINFO_METHOD_INFO* info)
+        private bool getMethodInfo(CORINFO_METHOD_STRUCT_* ftn, CORINFO_METHOD_INFO* info, CORINFO_CONTEXT_STRUCT* context)
         {
             MethodDesc method = HandleToObject(ftn);
```

Comment:
Do we need to take the exact context into account here as well?

Reply:
@jkotas yes, I was about to ask for help as I am a bit lost here, so for
```cs
bool Test() => EqualityComparer<string>.Default.Equals("str", "str");
```
when ILC compiles it, jit invokes `getMethodInfo` like this:

![image](https://github.com/dotnet/runtime/assets/523221/1117afca-08d8-499f-8249-11c188cf4090)

and I don't know how I can use the context to get a proper method info

Reply:
I tried `_compilation.TypeSystemContext.GetMethodForInstantiatedType(method, (InstantiatedType)ctxType)` but it complained that `method` should not be of `InstantiatedMethod`

Reply:
Perhaps https://github.com/dotnet/runtime/pull/88025/commits/1eddadb20d105e107cc28fca358632b2bcc00762